usaspending_api/accounts/v2/filters/account_download.py

```diff
@@ -400,8 +396,59 @@ def generate_federal_account_query(queryset, account_type, tas_id, filters):
         for val in values_dict["federal_account"]
         if val in all_summed_cols
     }
+    # These columns require a more complex aggregation than others
+    summed_cols["gross_outlay_amount"] = Sum(
+        generate_gross_outlay_amount_derived_field(account_type, closed_submission_queryset)
+    )
+    summed_cols["gross_outlay_amount_FYB_to_period_end"] = Sum(
+        generate_gross_outlay_amount_derived_field(account_type, closed_submission_queryset)
+    )
+    summed_cols["adjustments_to_unobligated_balance_brought_forward_cpe"] = Sum(
+        "adjustments_to_unobligated_balance_brought_forward_cpe"
+    )
+    queryset = queryset.annotate(**summed_cols)
 
-    return queryset.annotate(**summed_cols)
+    if account_type != "account_balances":
+        # If the file is not File A we are done
+        return queryset
+
+    # For File A, we expect one row per unique pair of federal_account_symbol and submission_period
+
+    # List of the columns that may be concatenated in File A
+    all_concat_cols = [
+        "owning_agency_name",
+        "reporting_agency_name",
+        "federal_account_name",
+        "agency_identifier_name",
+        "budget_function",
+        "budget_subfunction",
+    ]
+
+    # Perform a "GROUP BY" again to transform the data into the grain we expect
+    grouped_cols = [
+        fed_acct_values_dict[val]
+        for val in fed_acct_values_dict
+        if val not in all_summed_cols and val not in all_concat_cols
+    ]
+    queryset = queryset.values(*grouped_cols)
+
+    # Sum all summable fields again at new grain
+    # These columns are dynamic so Django cannot "SUM" them again
+    summed_cols["adjustments_to_unobligated_balance_brought_forward_cpe"] = F(
```

Comment:
For whatever reason, Django was throwing the following error when `SUM` was used again. That's why I opted for `F`. I've confirmed the column is still being totaled correctly.